be/src/exec/schema_scanner/schema_views_scanner.cpp

```diff
@@ -0,0 +1,227 @@
+        StringValue* str_slot = reinterpret_cast<StringValue*>(slot);
+        std::string db_name = SchemaHelper::extract_db_name(_db_result.dbs[_db_index - 1]);
+        str_slot->ptr = (char *)pool->allocate(db_name.size());
+        str_slot->len = db_name.size();
+        memcpy(str_slot->ptr, db_name.c_str(), str_slot->len);
+    }
+    // name
+    {
+        void *slot = tuple->get_slot(_tuple_desc->slots()[2]->tuple_offset());
+        StringValue* str_slot = reinterpret_cast<StringValue*>(slot);
+        const std::string* src = &tbl_status.name;
+        str_slot->len = src->length();
+        str_slot->ptr = (char *)pool->allocate(str_slot->len);
+        if (NULL == str_slot->ptr) {
+            return Status::InternalError("Allocate memcpy failed.");
+        }
+        memcpy(str_slot->ptr, src->c_str(), str_slot->len);
+    }
+    // definition
+    {
+        void *slot = tuple->get_slot(_tuple_desc->slots()[3]->tuple_offset());
+        StringValue* str_slot = reinterpret_cast<StringValue*>(slot);
+        const std::string* ddl_sql = &tbl_status.ddl_sql;
+        str_slot->len = ddl_sql->length();
+        str_slot->ptr = (char *)pool->allocate(str_slot->len);
+        if (NULL == str_slot->ptr) {
+            return Status::InternalError("Allocate memcpy failed.");
+        }
+        memcpy(str_slot->ptr, ddl_sql->c_str(), str_slot->len);
+    }
+    // check_option
+    {
+        void *slot = tuple->get_slot(_tuple_desc->slots()[4]->tuple_offset());
+        StringValue* str_slot = reinterpret_cast<StringValue*>(slot);
+        const std::string check_option = "NONE";
+        str_slot->len = check_option.length();
+        str_slot->ptr = (char *)pool->allocate(str_slot->len);
+        if (NULL == str_slot->ptr) {
+            return Status::InternalError("Allocate memcpy failed.");
+        }
+        memcpy(str_slot->ptr, check_option.c_str(), str_slot->len);
+    }
+    // is_updatable
+    {
+        void *slot = tuple->get_slot(_tuple_desc->slots()[5]->tuple_offset());
+        StringValue* str_slot = reinterpret_cast<StringValue*>(slot);
+        const std::string is_updatable = "YES";
+        str_slot->len = is_updatable.length();
+        str_slot->ptr = (char *)pool->allocate(str_slot->len);
+        if (NULL == str_slot->ptr) {
+            return Status::InternalError("Allocate memcpy failed.");
+        }
+        memcpy(str_slot->ptr, is_updatable.c_str(), str_slot->len);
+    }
+    // definer
+    {
+        void *slot = tuple->get_slot(_tuple_desc->slots()[6]->tuple_offset());
+        StringValue* str_slot = reinterpret_cast<StringValue*>(slot);
+        const std::string definer = "root@%";
+        str_slot->len = definer.length();
+        str_slot->ptr = (char *)pool->allocate(str_slot->len);
+        if (NULL == str_slot->ptr) {
+            return Status::InternalError("Allocate memcpy failed.");
+        }
+        memcpy(str_slot->ptr, definer.c_str(), str_slot->len);
```

Comment:
length always 6？

Reply:
this is column for table views in mysql, palo has no this information, we can only set a default one